Dalamud/Game/Gui/ContextMenu/CharacterData.cs

```diff
@@ -0,0 +1,193 @@
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+using Dalamud.Game.ClientState.Resolvers;
+using Dalamud.Memory;
+
+using FFXIVClientStructs.FFXIV.Client.UI.Info;
+
+using Lumina.Excel.GeneratedSheets;
+
+namespace Dalamud.Game.Gui.ContextMenu;
+
+/// <summary>
+/// Dalamud wrapper around a ClientStructs CharacterData.
+/// </summary>
+public unsafe class CharacterData
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CharacterData"/> class.
+    /// </summary>
+    /// <param name="data">Character data to wrap.</param>
+    internal CharacterData(InfoProxyCommonList.CharacterData data)
```

Comment:
It looks like you're copying all relevant data into this class. That's fine, but please document that you're doing this - our other wrappers normally read from the backing pointer as-is and expose an `Address` as an escape hatch for users that would like extra information from the raw struct.